### ISSUE TITLE
Track custom events

### ISSUE THREAD

**Issue by paaljoachim** [ibericode/koko-analytics#263]
Hey

I am wondering if a filter can be added to the plugin with some help to add custom events?

Let's say there was a kind of conditional to where one would like to add different kinds of events.
It would be nice to add additional events to the plugin as a new info box in the Analytics tracking screen.

**Comment by paaljoachim** [comment]
Heya Danny @dannyvankooten 

Can you add some information on how you added the custom events you mention?
I am thinking that I could then create a video tutorial + written tutorial on my own web site and/or on the Koko Analytics site.

---
(My initial comment was about...)
I was actually thinking about adding an UI to make it easier to add the custom events.
But having a how to for the various custom events you added would be very helpful. I just need the information from you and then I can explore it myself and create the documentation for it.

**Comment by dannyvankooten** [comment]
Hi @paaljoachim,

Sure thing; have a look at https://www.kokoanalytics.com/kb/tracking-events/ .

**Comment by paaljoachim** [comment]
-1-

I made a new event and called it "Device" for tracking which kind of device/screen size is in use. Such as mobile, tablet, desktop etc.

I then went to the Code Snippet plugin that I use and added this code:
https://github.com/ibericode/koko-analytics/blob/main/code-snippets/event-snippets/track-screen-width.js
Activated the code snippet in the footer.

Tested with my cell phone and a private tab on my desktop.
I noticed it did not show up.

So I went back to the documentation you wrote for track-screen-width.js 
Then I went to Koko Analytics Settings. 
Clicked the Device event listed under Event types.
Noticed tracking events codes.
Such as: `koko_analytics.trackEvent('Device');`

Went to the Code snippet I had earlier added.
In the track screen width code I used I changed the default: `koko_analytics.trackEvent('Screen width', bucket(screen.width));` 
to `
koko_analytics.trackEvent('Device', bucket(screen.width));`

So the device code now looks like this:

` window.addEventListener('load', function() {
    function bucket(n) {
        if (n < 500) {
            return '0 - 500';
        } else if (n < 1000) {
            return '500 - 1000';
        } else if (n < 1500) {
            return '1000 - 1500';
        } else if (n < 2000) {
            return '1500 - 2000';
        } else {
            return '2000+';
        }
    }

    koko_analytics.trackEvent('Device', bucket(screen.width));
});`

The code snippet plugin has a radio box I clicked to add the code into the footer.

I have a feeling I made a mistake someplace....or perhaps it takes a few hours before tracking information shows up.

![Screenshot 2024-11-23 at 09 30 46](https://github.com/user-attachments/assets/1ca3f9f8-1f7e-42e7-bf3d-27203ebd7735)

---------

-2-

Going back to settings I see it listed under Event types - Event name.
Why is Outbound link and Form submit mentioned there? As I have not activated these.
Should these be listed in a deactivated event types?

---

Thanks.

**Comment by dannyvankooten** [comment]
Nice work @paaljoachim. 

In order to make custom events a little easier, Koko Analytics Pro now includes several built-in custom events which can be easily enabled without having to write a single line of code. This applies to "Outbound link" and "Form submit". 

You think the registered event types list should show right away whether an event is enabled or disabled?

**Comment by paaljoachim** [comment]
Thank you @dannyvankooten Danny..:)

> You think the registered event types list should show right away whether an event is enabled or disabled?

From looking at the events list. I do not understand which are enabled or disabled. 
It would be nice with a simple method to enable or disable an event. 
This means I can not really answer your question.

I do not know how to enable "Outbound link" and "Form submit" events. Here I would assume that I would need to made another code snippet with the needed code I can likely find here: https://github.com/ibericode/koko-analytics/tree/main/code-snippets/event-snippets
It would be nice to also add a code snippet for Fluent Forms. It is a Forms plugin I am exploring these days.

Should additional registered events be made available? So the user can enable additional events.
Such as device/screen size (like I added).
Geo location.
Time on page 
and others?

When events are registered it seems like these by default are off and that one has to figure out a way to enable these.

**Comment by dannyvankooten** [comment]
![image](https://github.com/user-attachments/assets/ad874a23-e26e-4e0b-a27e-bd7b0b076e88)

@paaljoachim I'll see if I can remove the UI for those built-in events! You can enable or disable them right now through the radio inputs just above the events section.

**Comment by paaljoachim** [comment]
Ahh. I did not even notice the Track outbound link clicks and Track form submission as it was so different from the registered events UI. As if I did not expect to see it like that.

I believe I expected the registered events to also have a UI for enabling/disabling. 

So the procedure would be like so as I see it:
Events are registered by default or user registered. The question comes up why are not more common tracking registered events listed by default? As it seems likely the user would like to have additional events tracked. One could then disable or enable in the same place/list. Having the same registered list also be able to disable or enable events.
